--- gmp-5.0.2/configure.orig    2011-05-16 17:59:59.000000000 -0400             
+++ gmp-5.0.2/configure 2011-05-16 18:00:36.000000000 -0400                     
@@ -26446,8 +26446,8 @@                                                         
 esac
 
 cat >conftest.c <<EOF
-extern const int foo;		/* Suppresses C++'s suppression of foo */
-const int foo = {1,2,3};
+/* extern const int foo;		Suppresses C++'s suppression of foo */
+static const int foo = {1,2,3};
 EOF
 echo "Test program:" >&5
 cat conftest.c >&5
