diff -Nurd -x'*~' poppler-0.15.0.orig/glib/Makefile.in poppler-0.15.0/glib/Makefile.in
--- poppler-0.15.0.orig/glib/Makefile.in	2009-05-16 11:40:50.000000000 -0400
+++ poppler-0.15.0/glib/Makefile.in	2009-06-26 03:31:15.000000000 -0400
@@ -87,8 +87,6 @@
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
 libpoppler_glib_la_DEPENDENCIES =  \
-	$(top_builddir)/poppler/libpoppler.la \
-	$(top_builddir)/poppler/libpoppler-cairo.la \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 am__objects_1 = poppler-enums.lo
@@ -110,7 +108,6 @@
 @BUILD_WITH_GDK_TRUE@	test-poppler-glib.$(OBJEXT)
 test_poppler_glib_OBJECTS = $(am_test_poppler_glib_OBJECTS)
 @BUILD_WITH_GDK_TRUE@test_poppler_glib_DEPENDENCIES =  \
-@BUILD_WITH_GDK_TRUE@	$(top_builddir)/poppler/libpoppler.la \
 @BUILD_WITH_GDK_TRUE@	libpoppler-glib.la $(am__DEPENDENCIES_1) \
 @BUILD_WITH_GDK_TRUE@	$(am__DEPENDENCIES_1) \
 @BUILD_WITH_GDK_TRUE@	$(am__DEPENDENCIES_1)
@@ -406,10 +403,10 @@
 	-DG_LOG_DOMAIN=\"Poppler\"		\
 	-I$(top_srcdir)				\
 	-I$(top_srcdir)/poppler			\
+	$(FREETYPE_CFLAGS)			\
+	$(FONTCONFIG_CFLAGS)			\
 	$(POPPLER_GLIB_CFLAGS)			\
 	$(GDK_CFLAGS)				\
-	$(FREETYPE_CFLAGS)			\
-	$(FONTCONFIG_CFLAGS)                    \
 	$(POPPLER_GLIB_DISABLE_DEPRECATED)	\
 	$(POPPLER_GLIB_DISABLE_SINGLE_INCLUDES)
 
@@ -454,23 +451,22 @@
 	poppler-private.h
 
 libpoppler_glib_la_LIBADD = \
-	$(top_builddir)/poppler/libpoppler.la		\
-	$(top_builddir)/poppler/libpoppler-cairo.la	\
-	$(POPPLER_GLIB_LIBS)				\
-	$(GDK_LIBS)					\
+	-lpoppler					\
 	$(FREETYPE_LIBS)				\
-	$(FONTCONFIG_LIBS)
+	$(FONTCONFIG_LIBS)				\
+	$(POPPLER_GLIB_LIBS)				\
+	$(GDK_LIBS)
 
 libpoppler_glib_la_LDFLAGS = -version-info 6:0:0 @create_shared_lib@ @auto_import_flags@
 @BUILD_WITH_GDK_TRUE@test_poppler_glib_SOURCES = \
 @BUILD_WITH_GDK_TRUE@       test-poppler-glib.cc
 
 @BUILD_WITH_GDK_TRUE@test_poppler_glib_LDADD = \
-@BUILD_WITH_GDK_TRUE@	$(top_builddir)/poppler/libpoppler.la	\
+@BUILD_WITH_GDK_TRUE@	-lpoppler	\
 @BUILD_WITH_GDK_TRUE@	libpoppler-glib.la			\
+@BUILD_WITH_GDK_TRUE@   $(FREETYPE_LIBS)			\
 @BUILD_WITH_GDK_TRUE@	$(POPPLER_GLIB_LIBS)			\
-@BUILD_WITH_GDK_TRUE@	$(GDK_LIBS)				\
-@BUILD_WITH_GDK_TRUE@	$(FREETYPE_LIBS)
+@BUILD_WITH_GDK_TRUE@	$(GDK_LIBS)
 
 BUILT_SOURCES = \
 	poppler-enums.c				\
diff -Nurd -x'*~' poppler-0.15.0.orig/glib/demo/Makefile.in poppler-0.15.0/glib/demo/Makefile.in
--- poppler-0.15.0.orig/glib/demo/Makefile.in	2009-05-16 11:40:50.000000000 -0400
+++ poppler-0.15.0/glib/demo/Makefile.in	2009-06-26 03:31:15.000000000 -0400
@@ -65,7 +65,7 @@
 am__DEPENDENCIES_1 =
 poppler_glib_demo_DEPENDENCIES =  \
 	$(top_builddir)/glib/libpoppler-glib.la \
-	$(top_builddir)/poppler/libpoppler.la $(am__DEPENDENCIES_1)
+	$(am__DEPENDENCIES_1)
 AM_V_lt = $(am__v_lt_$(V))
 am__v_lt_ = $(am__v_lt_$(AM_DEFAULT_VERBOSITY))
 am__v_lt_0 = --silent
@@ -360,7 +360,7 @@
 
 LDADD = \
 	$(top_builddir)/glib/libpoppler-glib.la	\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler \
 	$(GTK_TEST_LIBS)
 
 all: all-am
diff -Nurd -x'*~' poppler-0.15.0.orig/glib/reference/Makefile.in poppler-0.15.0/glib/reference/Makefile.in
--- poppler-0.15.0.orig/glib/reference/Makefile.in	2009-05-16 11:40:50.000000000 -0400
+++ poppler-0.15.0/glib/reference/Makefile.in	2009-06-26 03:31:15.000000000 -0400
@@ -331,15 +331,15 @@
 # e.g. INCLUDES=-I$(top_srcdir) -I$(top_builddir) $(GTK_DEBUG_FLAGS)
 # e.g. GTKDOC_LIBS=$(top_builddir)/gtk/$(gtktargetlib)
 INCLUDES = -I$(top_srcdir)/glib -I$(top_builddir)/glib	\
+	$(FREETYPE_CFLAGS) \
 	$(CAIRO_CFLAGS)					\
 	$(POPPLER_GLIB_CFLAGS)				\
-	$(GDK_CFLAGS)					\
-	$(FREETYPE_CFLAGS)
+	$(GDK_CFLAGS)					
 
-GTKDOC_LIBS = $(top_builddir)/poppler/libpoppler.la	\
+GTKDOC_LIBS = -lpoppler \
 	$(top_builddir)/glib/libpoppler-glib.la		\
-	$(POPPLER_GLIB_LIBS)				\
 	$(FREETYPE_LIBS)				\
+	$(POPPLER_GLIB_LIBS)				\
 	$(cairo_libs)
 
 @GTK_DOC_USE_LIBTOOL_FALSE@GTKDOC_CC = $(CC) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
diff -Nurd -x'*~' poppler-0.15.0.orig/poppler/Makefile.in poppler-0.15.0/poppler/Makefile.in
--- poppler-0.15.0.orig/poppler/Makefile.in	2009-05-16 11:40:50.000000000 -0400
+++ poppler-0.15.0/poppler/Makefile.in	2009-06-26 03:31:15.000000000 -0400
@@ -527,6 +527,8 @@
 INCLUDES = \
 	-I$(top_srcdir)				\
 	-I$(top_srcdir)/goo			\
+	$(FREETYPE_CFLAGS)                      \
+	$(FONTCONFIG_CFLAGS)                    \
 	$(cms_includes)				\
 	$(splash_includes)			\
 	$(cairo_includes)			\
@@ -544,6 +546,8 @@
 libpoppler_la_LIBADD = \
 	$(top_builddir)/goo/libgoo.la		\
 	$(top_builddir)/fofi/libfofi.la		\
+	$(FREETYPE_LIBS)                        \
+	$(FONTCONFIG_LIBS)                      \
 	$(cms_libs)				\
 	$(splash_libs)				\
 	$(libjpeg_libs)				\
@@ -552,8 +556,6 @@
 	$(libcurl_libs)				\
 	$(libjpeg2000_libs)			\
 	$(abiword_libs)				\
-	$(FREETYPE_LIBS)			\
-	$(FONTCONFIG_LIBS)			\
 	$(PTHREAD_LIBS)				\
 	$(win32_libs)
 
diff -Nurd -x'*~' poppler-0.15.0.orig/poppler-cairo.pc.cmake poppler-0.15.0/poppler-cairo.pc.cmake
--- poppler-0.15.0.orig/poppler-cairo.pc.cmake	2009-05-02 08:14:56.000000000 -0400
+++ poppler-0.15.0/poppler-cairo.pc.cmake	2011-01-05 16:55:56.000000000 -0500
@@ -6,4 +6,4 @@
 Name: poppler-cairo
 Description: Cairo backend for Poppler PDF rendering library
 Version: @POPPLER_VERSION@
-Requires: poppler = @POPPLER_VERSION@ cairo >= @CAIRO_VERSION@
+Requires: poppler = @POPPLER_VERSION@
diff -Nurd -x'*~' poppler-0.15.0.orig/poppler-cairo.pc.in poppler-0.15.0/poppler-cairo.pc.in
--- poppler-0.15.0.orig/poppler-cairo.pc.in	2009-05-02 08:14:56.000000000 -0400
+++ poppler-0.15.0/poppler-cairo.pc.in	2011-01-05 16:56:11.000000000 -0500
@@ -6,4 +6,4 @@
 Name: poppler-cairo
 Description: Cairo backend for Poppler PDF rendering library
 Version: @VERSION@
-Requires: poppler = @VERSION@ cairo >= @CAIRO_VERSION@
+Requires: poppler = @VERSION@
diff -Nurd -x'*~' poppler-0.15.0.orig/qt/Makefile.in poppler-0.15.0/qt/Makefile.in
--- poppler-0.15.0.orig/qt/Makefile.in	2009-05-16 11:40:50.000000000 -0400
+++ poppler-0.15.0/qt/Makefile.in	2009-06-26 03:31:15.000000000 -0400
@@ -80,7 +80,7 @@
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
-libpoppler_qt_la_DEPENDENCIES = $(top_builddir)/poppler/libpoppler.la \
+libpoppler_qt_la_DEPENDENCIES = \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
 am_libpoppler_qt_la_OBJECTS = poppler-document.lo poppler-fontinfo.lo \
@@ -97,7 +97,7 @@
 PROGRAMS = $(noinst_PROGRAMS)
 am_test_poppler_qt_OBJECTS = test-poppler-qt.$(OBJEXT)
 test_poppler_qt_OBJECTS = $(am_test_poppler_qt_OBJECTS)
-test_poppler_qt_DEPENDENCIES = $(top_builddir)/poppler/libpoppler.la \
+test_poppler_qt_DEPENDENCIES = \
 	libpoppler-qt.la $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir) -I$(top_builddir)/poppler
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -346,10 +346,10 @@
 INCLUDES = \
 	-I$(top_srcdir)				\
 	-I$(top_srcdir)/poppler			\
+	$(FREETYPE_CFLAGS)                      \
+	$(FONTCONFIG_CFLAGS)                    \
 	$(cairo_includes)			\
-	$(libqt3_CFLAGS)			\
-	$(FREETYPE_CFLAGS)			\
-	$(FONTCONFIG_CFLAGS)
+	$(libqt3_CFLAGS)
 
 poppler_includedir = $(includedir)/poppler/qt3
 poppler_include_HEADERS = \
@@ -369,20 +369,20 @@
 	poppler-private.h
 
 libpoppler_qt_la_LIBADD = \
-	$(top_builddir)/poppler/libpoppler.la	\
-	$(POPPLER_QT_LIBS)			\
+	-lpoppler				\
 	$(FREETYPE_LIBS)			\
-	$(FONTCONFIG_LIBS)
+	$(FONTCONFIG_LIBS)			\
+	$(POPPLER_QT_LIBS)
 
 libpoppler_qt_la_LDFLAGS = -version-info 2:0:0 @create_shared_lib@
 test_poppler_qt_SOURCES = \
        test-poppler-qt.cpp
 
 test_poppler_qt_LDADD = \
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpopler				\
 	libpoppler-qt.la			\
-	$(libqt3_LIBS)				\
-	$(FREETYPE_LIBS)
+	$(FREETYPE_LIBS)			\
+	$(libqt3_libs)
 
 all: all-am
 
diff -Nurd -x'*~' poppler-0.15.0.orig/qt4/demos/Makefile.in poppler-0.15.0/qt4/demos/Makefile.in
--- poppler-0.15.0.orig/qt4/demos/Makefile.in	2009-05-16 11:40:50.000000000 -0400
+++ poppler-0.15.0/qt4/demos/Makefile.in	2009-06-26 03:31:15.000000000 -0400
@@ -61,7 +61,7 @@
 	toc.$(OBJEXT) viewer.$(OBJEXT)
 poppler_qt4viewer_OBJECTS = $(am_poppler_qt4viewer_OBJECTS)
 am__DEPENDENCIES_1 =
-am__DEPENDENCIES_2 = $(top_builddir)/poppler/libpoppler.la \
+am__DEPENDENCIES_2 = \
 	$(top_builddir)/qt4/src/libpoppler-qt4.la \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 poppler_qt4viewer_DEPENDENCIES = $(am__DEPENDENCIES_2)
@@ -321,7 +321,7 @@
 	$(POPPLER_QT4_CFLAGS)
 
 LDADDS = \
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler \
 	$(top_builddir)/qt4/src/libpoppler-qt4.la	\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
diff -Nurd -x'*~' poppler-0.15.0.orig/qt4/src/Makefile.in poppler-0.15.0/qt4/src/Makefile.in
--- poppler-0.15.0.orig/qt4/src/Makefile.in	2009-05-16 11:40:50.000000000 -0400
+++ poppler-0.15.0/qt4/src/Makefile.in	2009-06-26 03:31:15.000000000 -0400
@@ -79,8 +79,6 @@
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
 libpoppler_qt4_la_DEPENDENCIES =  \
-	$(top_builddir)/poppler/libpoppler.la \
-	$(top_builddir)/poppler/libpoppler-arthur.la \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 am_libpoppler_qt4_la_OBJECTS = libpoppler_qt4_la-poppler-document.lo \
 	libpoppler_qt4_la-poppler-page.lo \
@@ -405,8 +403,7 @@
 	-Dpoppler_qt4_EXPORTS
 
 libpoppler_qt4_la_LIBADD = \
-	$(top_builddir)/poppler/libpoppler.la	\
-	$(top_builddir)/poppler/libpoppler-arthur.la	\
+	-lpoppler	\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
 
diff -Nurd -x'*~' poppler-0.15.0.orig/qt4/tests/Makefile.in poppler-0.15.0/qt4/tests/Makefile.in
--- poppler-0.15.0.orig/qt4/tests/Makefile.in	2009-05-16 11:40:51.000000000 -0400
+++ poppler-0.15.0/qt4/tests/Makefile.in	2009-06-26 03:31:15.000000000 -0400
@@ -86,7 +86,7 @@
 @BUILD_POPPLER_QT4_TRUE@	check_actualtext.$(OBJEXT)
 check_actualtext_OBJECTS = $(am_check_actualtext_OBJECTS)
 am__DEPENDENCIES_1 =
-am__DEPENDENCIES_2 = $(top_builddir)/poppler/libpoppler.la \
+am__DEPENDENCIES_2 = \
 	$(top_builddir)/qt4/src/libpoppler-qt4.la \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 @BUILD_POPPLER_QT4_TRUE@check_actualtext_DEPENDENCIES =  \
@@ -456,7 +456,7 @@
 	$(POPPLER_QT4_CFLAGS)
 
 LDADDS = \
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler \
 	$(top_builddir)/qt4/src/libpoppler-qt4.la	\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
diff -Nurd -x'*~' poppler-0.15.0.orig/test/Makefile.in poppler-0.15.0/test/Makefile.in
--- poppler-0.15.0.orig/test/Makefile.in	2009-05-16 11:40:51.000000000 -0400
+++ poppler-0.15.0/test/Makefile.in	2009-06-26 03:31:15.000000000 -0400
@@ -331,8 +331,8 @@
 @BUILD_CAIRO_OUTPUT_TRUE@@BUILD_GTK_TEST_TRUE@	pdf_inspector
 
 @BUILD_CAIRO_OUTPUT_TRUE@@BUILD_GTK_TEST_TRUE@cairo_includes = \
-@BUILD_CAIRO_OUTPUT_TRUE@@BUILD_GTK_TEST_TRUE@	$(CAIRO_CFLAGS)				\
-@BUILD_CAIRO_OUTPUT_TRUE@@BUILD_GTK_TEST_TRUE@	$(FREETYPE_CFLAGS)
+@BUILD_CAIRO_OUTPUT_TRUE@@BUILD_GTK_TEST_TRUE@	$(FREETYPE_CFLAGS) \
+@BUILD_CAIRO_OUTPUT_TRUE@@BUILD_GTK_TEST_TRUE@	$(CAIRO_CFLAGS)				
 
 @BUILD_SPLASH_OUTPUT_TRUE@perf_test = \
 @BUILD_SPLASH_OUTPUT_TRUE@	perf-test
@@ -345,9 +345,9 @@
 	-I$(top_srcdir)/poppler			\
 	-I$(top_srcdir)/glib			\
 	-I$(top_builddir)/glib			\
+	$(FONTCONFIG_CFLAGS) \
 	$(cairo_includes)			\
-	$(GTK_TEST_CFLAGS)			\
-	$(FONTCONFIG_CFLAGS)
+	$(GTK_TEST_CFLAGS)			
 
 gtk_splash_test_SOURCES = \
        gtk-splash-test.cc
@@ -363,9 +363,9 @@
 gtk_cairo_test_LDADD = \
 	$(top_builddir)/poppler/libpoppler.la		\
 	$(top_builddir)/glib/libpoppler-glib.la		\
+	$(FONTCONFIG_LIBS) \
 	$(CAIRO_LIBS)					\
-	$(GTK_TEST_LIBS)				\
-	$(FONTCONFIG_LIBS)
+	$(GTK_TEST_LIBS)				
 
 pdf_inspector_SOURCES = \
        pdf-inspector.cc
@@ -373,8 +373,8 @@
 pdf_inspector_LDADD = \
 	$(top_builddir)/poppler/libpoppler-cairo.la	\
 	$(top_builddir)/poppler/libpoppler.la		\
-	$(CAIRO_LIBS)					\
 	$(FREETYPE_LIBS)				\
+	$(CAIRO_LIBS)					\
 	$(GTK_TEST_LIBS)				\
 	$(X_EXTRA_LIBS)
 
diff -Nurd -x'*~' poppler-0.15.0.orig/utils/Makefile.in poppler-0.15.0/utils/Makefile.in
--- poppler-0.15.0.orig/utils/Makefile.in	2009-05-16 11:40:51.000000000 -0400
+++ poppler-0.15.0/utils/Makefile.in	2009-06-26 03:31:15.000000000 -0400
@@ -397,14 +397,14 @@
 	-I$(top_srcdir)				\
 	-I$(top_srcdir)/utils			\
 	-I$(top_srcdir)/poppler			\
-	$(UTILS_CFLAGS)				\
 	$(FONTCONFIG_CFLAGS)			\
+	$(UTILS_CFLAGS)				\
 	$(ABIWORD_CFLAGS)
 
 LDADD = \
 	$(top_builddir)/poppler/libpoppler.la	\
-	$(UTILS_LIBS)				\
-	$(FONTCONFIG_LIBS)
+	$(FONTCONFIG_LIBS) \
+	$(UTILS_LIBS)
 
 @BUILD_ABIWORD_OUTPUT_TRUE@pdftoabw_SOURCES = \
 @BUILD_ABIWORD_OUTPUT_TRUE@	pdftoabw.cc				\
diff -Nurd -x'*~' poppler-0.15.0.orig/xpdfrc.poppler4 poppler-0.15.0/xpdfrc.poppler4
--- poppler-0.15.0.orig/xpdfrc.poppler4	1969-12-31 19:00:00.000000000 -0500
+++ poppler-0.15.0/xpdfrc.poppler4	2009-06-26 03:31:15.000000000 -0400
@@ -0,0 +1,91 @@
+#========================================================================
+#
+# Sample xpdfrc file
+#
+# The Xpdf tools look for a config file in two places:
+# 1. ~/.xpdfrc
+# 2. in a system-wide directory, typically /usr/local/etc/xpdfrc
+#
+# This sample config file demonstrates some of the more common
+# configuration options.  Everything here is commented out.  You
+# should edit things (especially the file/directory paths, since
+# they'll likely be different on your system), and uncomment whichever
+# options you want to use.  For complete details on config file syntax
+# and available options, please see the xpdfrc(5) man page.
+#
+# Also, the Xpdf language support packages each include a set of
+# options to be added to the xpdfrc file.
+#
+# http://www.foolabs.com/xpdf/
+#
+#========================================================================
+
+#----- display fonts
+
+# These map the Base-14 fonts to the Type 1 fonts that ship with
+# ghostscript.  You'll almost certainly want to use something like
+# this, but you'll need to adjust this to point to wherever
+# ghostscript is installed on your system.  (But if the fonts are
+# installed in a "standard" location, xpdf will find them
+# automatically.)
+
+displayFontT1 Times-Roman              @PREFIX@/share/ghostscript/fonts/n021003l.pfb
+displayFontT1 Times-Italic             @PREFIX@/share/ghostscript/fonts/n021023l.pfb
+displayFontT1 Times-Bold               @PREFIX@/share/ghostscript/fonts/n021004l.pfb
+displayFontT1 Times-BoldItalic         @PREFIX@/share/ghostscript/fonts/n021024l.pfb
+displayFontT1 Helvetica                @PREFIX@/share/ghostscript/fonts/n019003l.pfb
+displayFontT1 Helvetica-Oblique        @PREFIX@/share/ghostscript/fonts/n019023l.pfb
+displayFontT1 Helvetica-Bold           @PREFIX@/share/ghostscript/fonts/n019004l.pfb
+displayFontT1 Helvetica-BoldOblique    @PREFIX@/share/ghostscript/fonts/n019024l.pfb
+displayFontT1 Courier                  @PREFIX@/share/ghostscript/fonts/n022003l.pfb
+displayFontT1 Courier-Oblique          @PREFIX@/share/ghostscript/fonts/n022023l.pfb
+displayFontT1 Courier-Bold             @PREFIX@/share/ghostscript/fonts/n022004l.pfb
+displayFontT1 Courier-BoldOblique      @PREFIX@/share/ghostscript/fonts/n022024l.pfb
+displayFontT1 Symbol                   @PREFIX@/share/ghostscript/fonts/s050000l.pfb
+displayFontT1 ZapfDingbats             @PREFIX@/share/ghostscript/fonts/d050000l.pfb
+
+# If you need to display PDF files that refer to non-embedded fonts,
+# you should add one or more fontDir options to point to the
+# directories containing the font files.  Xpdf will only look at .pfa,
+# .pfb, and .ttf files in those directories (other files will simply
+# be ignored).
+
+fontDir                /usr/local/fonts/bakoma
+
+#----- PostScript output control
+
+# Set the default PostScript file or command.
+
+psFile                 "|lpr -Pmyprinter"
+
+# Set the default PostScript paper size -- this can be letter, legal,
+# A4, or A3.  You can also specify a paper size as width and height
+# (in points).
+
+psPaperSize            letter
+
+#----- text output control
+
+# Choose a text encoding for copy-and-paste and for pdftotext output.
+# The Latin1, ASCII7, and UTF-8 encodings are built into Xpdf.  Other
+# encodings are available in the language support packages.
+
+textEncoding           UTF-8
+
+# Choose the end-of-line convention for multi-line copy-and-past and
+# for pdftotext output.  The available options are unix, mac, and dos.
+
+textEOL                unix
+
+#----- misc settings
+
+# Enable t1lib, FreeType, and anti-aliased text.
+
+enableT1lib            yes
+enableFreeType         yes
+antialias              yes
+
+# Set the command used to run a web browser when a URL hyperlink is
+# clicked.
+
+urlCommand     "open 'openURL(%s)'"
