diff -Nurd octave-3.2.4/src/DLD-FUNCTIONS/__magick_read__.cc octave-3.2.4.patched/src/DLD-FUNCTIONS/__magick_read__.cc
--- octave-3.2.4/src/DLD-FUNCTIONS/__magick_read__.cc	2009-08-25 04:26:01.000000000 -0400
+++ octave-3.2.4.patched/src/DLD-FUNCTIONS/__magick_read__.cc	2011-09-22 13:21:51.000000000 -0400
@@ -34,6 +34,7 @@
 #ifdef HAVE_MAGICK
 
 #include <GraphicsMagick/Magick++.h>
+using namespace Magick;
 
 unsigned int
 scale_quantum_to_depth (const Magick::Quantum& quantum, unsigned int depth)
@@ -357,6 +358,8 @@
   octave_value_list output;
 
 #ifdef HAVE_MAGICK
+  // Initialize GraphicsMagick
+  InitializeMagick(NULL);
 
   if (args.length () > 2 || args.length () < 1 || ! args(0).is_string ()
       || nargout > 3)
@@ -767,6 +770,8 @@
 @end deftypefn")
 {
   octave_value_list retval;
+  //Initialize GraphicsMagick
+  InitializeMagick(NULL);
 
 #ifdef HAVE_MAGICK
   int nargin = args.length ();
@@ -897,6 +902,8 @@
   octave_value_list output;
 
 #ifdef HAVE_MAGICK
+  // Initialize GraphicsMagick
+  InitializeMagick(NULL);
 
   if (args.length () < 1 || ! args (0).is_string ())
     {
