? t/basepath/src
Index: bootstrap
===================================================================
RCS file: /cvsroot/fink/fink/bootstrap,v
retrieving revision 1.44
diff -u -b -w -r1.44 bootstrap
--- bootstrap	21 Mar 2011 20:42:35 -0000	1.44
+++ bootstrap	26 Mar 2011 18:02:08 -0000
@@ -532,10 +532,14 @@
 
 my $perlexe  = "/usr/bin/perl";
 
-if ($distribution gt "10.5") { 
+if ($distribution eq "10.6") { 
 	$perlexe = "/usr/bin/arch -arch $arch /usr/bin/perl5.10.0";
 }
 
+if ($distribution gt "10.6") {
+        $perlexe = "/usr/bin/arch -arch $arch /usr/bin/perl5.12";
+}
+
 if (($arch eq "x86_64") and ($distribution eq "10.5")) {
 	$perlexe = "$bsbase/bin/perl5.8.8";
 }
Index: setup.sh
===================================================================
RCS file: /cvsroot/fink/fink/setup.sh,v
retrieving revision 1.42
diff -u -b -w -r1.42 setup.sh
--- setup.sh	15 Jan 2011 22:32:51 -0000	1.42
+++ setup.sh	26 Mar 2011 18:02:08 -0000
@@ -42,10 +42,14 @@
   fi
 fi
 
-if [ $osMajorVer -gt 9 ]; then
+if [ $osMajorVer -eq 10 ]; then
   perlexe="/usr/bin/arch -arch $architecture /usr/bin/perl5.10.0"
 fi
 
+if [ $osMajorVer -gt 10 ]; then
+  perlexe="/usr/bin/arch -arch $architecture /usr/bin/perl5.12"
+fi
+
 echo "Creating $fink..." 
 sed -e "s|@BASEPATH@|$basepath|g" -e "s|@PERLEXE@|$perlexe|g" < fink.in > fink
 
Index: perlmod/Fink/Bootstrap.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Bootstrap.pm,v
retrieving revision 1.199
diff -u -b -w -r1.199 Bootstrap.pm
--- perlmod/Fink/Bootstrap.pm	21 Mar 2011 20:42:36 -0000	1.199
+++ perlmod/Fink/Bootstrap.pm	26 Mar 2011 18:02:08 -0000
@@ -129,7 +129,6 @@
 	my $bootstrap = shift @_ || 0;
 	my $arch = shift @_ || "";
 	my ($distribution, $gcc, $build, $transitional);
-
 	# We test for an obsolete version of gcc3.3, and refuse to proceed if
 	# it is present.
 	#
@@ -281,6 +280,12 @@
 			"of Mac OS X might work with Fink, but there are no " .
 			"guarantees.");
 		$distribution = "10.6";
+        } elsif ($host =~ /^i386-apple-darwin11\./) {
+                &print_breaking("This system was not released at the time " .
+                        "this Fink release was made.  Prerelease versions " .
+                        "of Mac OS X might work with Fink, but there are no " .
+                        "guarantees.");
+                $distribution = "10.7";
 	} elsif ($host =~ /^powerpc-apple-darwin1\.[0-2]/) {
 		&print_breaking("This system is outdated and not supported ".
 			"by this Fink release. Please update to Mac OS X ".
@@ -455,6 +460,7 @@
 	} elsif ("$]" == "5.008006") {
 	} elsif ("$]" == "5.008008") {
 	} elsif ("$]" == "5.010000") {
+        } elsif ("$]" == "5.012003") {
 	} else {
 # unsupported version of perl
 		$perl_is_supported = 0;
@@ -1061,7 +1067,8 @@
 		"10.3" => "10.3",
 		"10.4" => "10.4",
 		"10.5" => "10.4",
-		"10.6" => "10.4"
+		"10.6" => "10.4",
+                "10.7" => "10.4"
 		);
 
 	return $selfupdatetrees{$distribution};
Index: perlmod/Fink/Checksum.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Checksum.pm,v
retrieving revision 1.19
diff -u -b -w -r1.19 Checksum.pm
--- perlmod/Fink/Checksum.pm	15 Jan 2011 22:48:55 -0000	1.19
+++ perlmod/Fink/Checksum.pm	26 Mar 2011 18:02:08 -0000
@@ -24,7 +24,7 @@
 
 use Fink::Config qw($config);
 use Fink::Services qw(&find_subpackages);
-use UNIVERSAL qw(isa);
+# use UNIVERSAL qw(isa);
 
 BEGIN {
 	use Exporter ();
@@ -115,7 +115,7 @@
 
 	if ($@) {
 		die "unable to load checksum plugin ($plugin): $@\n";
-	} elsif (not isa($self, "Fink::Checksum")) {
+	} elsif (not $self->isa("Fink::Checksum")) {
 		die "unknown checksum plugin: $plugin\n";
 	}
 
Index: perlmod/Fink/Services.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Services.pm,v
retrieving revision 1.232
diff -u -b -w -r1.232 Services.pm
--- perlmod/Fink/Services.pm	15 Jan 2011 22:48:55 -0000	1.232
+++ perlmod/Fink/Services.pm	26 Mar 2011 18:02:08 -0000
@@ -1325,7 +1325,7 @@
 # Note: we no longer support 10.1 or 10.2-gcc3.1 in fink, we don't
 # specify default values for these.
 
-	my %system_gcc_default = ('10.2' => '3.3', '10.3' => '3.3', '10.4' => '4.0', '10.5' => '4.0', '10.6' => '4.2');
+	my %system_gcc_default = ('10.2' => '3.3', '10.3' => '3.3', '10.4' => '4.0', '10.5' => '4.0', '10.6' => '4.2', '10.7' => '4.2');
 	my %gcc_abi_default = ('2.95' => '2.95', '3.1' => '3.1', '3.3' => '3.3', '4.0' => '3.3', '4.2' => '3.3');
 
 	if (my $sw_vers = get_osx_vers_long())
@@ -1422,7 +1422,7 @@
 
 sub get_darwin_equiv
 {
-	my %darwin_osx = ('1' => '10.0', '5' => '10.1', '6' => '10.2', '7' => '10.3', '8' => '10.4', '9' => '10.5', '10' => '10.6');
+	my %darwin_osx = ('1' => '10.0', '5' => '10.1', '6' => '10.2', '7' => '10.3', '8' => '10.4', '9' => '10.5', '10' => '10.6', '11' => '10.7');
 	return $darwin_osx{get_kernel_vers()};
 }
 
