--- freeglut-3.0.0-rc1/progs/demos/shapes/shapes.c.orig	2014-11-03 22:10:17.000000000 -0500
+++ freeglut-3.0.0-rc1/progs/demos/shapes/shapes.c	2014-11-03 22:10:34.000000000 -0500
@@ -39,6 +39,7 @@
     \include demos/shapes/shapes.c
 */
 
+#include <GL/glew.h>
 #include <GL/freeglut.h>
 
 #include <stdarg.h>
--- freeglut-3.0.0-rc1/progs/demos/Fractals_random/fractals_random.c.orig	2014-10-29 13:09:57.000000000 -0400
+++ freeglut-3.0.0-rc1/progs/demos/Fractals_random/fractals_random.c	2014-11-03 22:16:17.000000000 -0500
@@ -20,6 +20,7 @@
  *  Escape - quit
  */
 
+#include <GL/glew.h>
 #include <GL/freeglut.h>
 #include <stdio.h>
 #include <stdlib.h>
--- freeglut-3.0.0-rc1/progs/demos/smooth_opengl3/smooth_opengl3.c.orig	2014-11-03 22:49:18.000000000 -0500
+++ freeglut-3.0.0-rc1/progs/demos/smooth_opengl3/smooth_opengl3.c	2014-11-03 22:49:43.000000000 -0500
@@ -42,6 +42,7 @@
  *
  */
 
+#include <GL/glew.h>
 #include <GL/freeglut.h>
 #include <stdlib.h>
 #include <stdio.h>
--- freeglut-3.0.0-rc1/progs/demos/Resizer/Resizer.cpp.orig	2014-11-03 22:53:02.000000000 -0500
+++ freeglut-3.0.0-rc1/progs/demos/Resizer/Resizer.cpp	2014-11-03 22:53:36.000000000 -0500
@@ -1,6 +1,7 @@
-#include <stdio.h>
+#include <GL/glew.h>
 
 #include <GL/freeglut.h>
+#include <stdio.h>
 
 int nWindow, nChildWindow = -1;
 int nLoopMain = 0;
diff -Nurd -x'*~' freeglut-3.0.0-rc2.orig/CMakeLists.txt freeglut-3.0.0-rc2/CMakeLists.txt
--- freeglut-3.0.0-rc2.orig/CMakeLists.txt	2014-12-25 09:35:09.000000000 -0500
+++ freeglut-3.0.0-rc2/CMakeLists.txt	2015-01-05 02:11:03.000000000 -0500
@@ -1,6 +1,9 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8 FATAL_ERROR)
+CMAKE_MINIMUM_REQUIRED(VERSION 2.8.8 FATAL_ERROR)
 PROJECT(freeglut)
 
+# for multiarch LIBDIR support (requires cmake>=2.8.8)
+INCLUDE(GNUInstallDirs)
+
 # NOTE: On Windows and Cygwin, the dll's are placed in the
 # CMAKE_RUNTIME_OUTPUT_DIRECTORY, while their corresponding import
 # libraries end up in CMAKE_ARCHIVE_OUTPUT_DIRECTORY. On other
@@ -413,8 +416,8 @@
 IF(FREEGLUT_BUILD_SHARED_LIBS)
     INSTALL(TARGETS freeglut
             RUNTIME DESTINATION bin
-            LIBRARY DESTINATION lib
-            ARCHIVE DESTINATION lib
+            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     )
     IF(INSTALL_PDB)
         INSTALL(FILES ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/Debug/freeglut${CMAKE_DEBUG_POSTFIX}.pdb
@@ -425,8 +428,8 @@
 IF(FREEGLUT_BUILD_STATIC_LIBS)
     INSTALL(TARGETS freeglut_static
             RUNTIME DESTINATION bin
-            LIBRARY DESTINATION lib
-            ARCHIVE DESTINATION lib
+            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     )
     IF(INSTALL_PDB)
         INSTALL(FILES ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/Debug/freeglut_static${CMAKE_DEBUG_POSTFIX}.pdb
diff -Nurd -x'*~' freeglut-3.0.0-rc2.orig/freeglut.pc.in freeglut-3.0.0-rc2/freeglut.pc.in
--- freeglut-3.0.0-rc2.orig/freeglut.pc.in	2012-04-02 18:22:33.000000000 -0400
+++ freeglut-3.0.0-rc2/freeglut.pc.in	2015-01-05 02:27:32.000000000 -0500
@@ -1,5 +1,5 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-libdir=${prefix}/lib
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 includedir=${prefix}/include
 
 Name: glut
