diff -uNr cfe-3.8.0.src.orig/lib/Driver/Tools.cpp cfe-3.8.0.src/lib/Driver/Tools.cpp
--- cfe-3.8.0.src.orig/lib/Driver/Tools.cpp	2015-09-21 08:53:06.000000000 -0400
+++ cfe-3.8.0.src/lib/Driver/Tools.cpp	2015-09-21 09:46:20.000000000 -0400
@@ -6712,12 +6712,18 @@
                    options::OPT_fno_openmp, false)) {
     switch (getOpenMPRuntime(getToolChain(), Args)) {
     case OMPRT_OMP:
+      // Help clang find libomp.dylib
+      CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-@BRV@/lib");
       CmdArgs.push_back("-lomp");
       break;
     case OMPRT_GOMP:
+      // Help clang find libgomp.dylib
+      CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-@BRV@/lib");
       CmdArgs.push_back("-lgomp");
       break;
     case OMPRT_IOMP5:
+      // Help clang find libiomp5.dylib
+      CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-@BRV@/lib");
       CmdArgs.push_back("-liomp5");
       break;
     case OMPRT_Unknown:
@@ -8412,9 +8418,13 @@
         // Also link the particular OpenMP runtimes.
         switch (getOpenMPRuntime(ToolChain, Args)) {
         case OMPRT_OMP:
+	  // Help clang find libomp.dylib
+	  CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-@BRV@/lib");
           CmdArgs.push_back("-lomp");
           break;
         case OMPRT_GOMP:
+          // Help clang find libgomp.dylib
+          CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-@BRV@/lib");
           CmdArgs.push_back("-lgomp");
 
           // FIXME: Exclude this for platforms with libgomp that don't require
@@ -8422,6 +8432,8 @@
           CmdArgs.push_back("-lrt");
           break;
         case OMPRT_IOMP5:
+	  // Help clang find libiomp5.dylib
+	  CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-@BRV@/lib");
           CmdArgs.push_back("-liomp5");
           break;
         case OMPRT_Unknown:
