--- cfe-3.7.0.src/lib/Driver/Tools.cpp.orig	2015-05-27 23:09:32.000000000 -0400
+++ cfe-3.7.0.src/lib/Driver/Tools.cpp	2015-05-27 23:38:44.000000000 -0400
@@ -6366,6 +6366,8 @@
       CmdArgs.push_back("-lgomp");
       break;
     case OMPRT_IOMP5:
+      // Help clang find libiomp5.dylib
+      CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-@BRV@/lib");
       CmdArgs.push_back("-liomp5");
       break;
     case OMPRT_Unknown:
@@ -8100,6 +8102,8 @@
           CmdArgs.push_back("-lrt");
           break;
         case OMPRT_IOMP5:
+	  // Help clang find libiomp5.dylib
+	  CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-@BRV@/lib");
           CmdArgs.push_back("-liomp5");
           break;
         case OMPRT_Unknown:
