--- polly-3.6.0.src/lib/CMakeLists.txt.orig	2015-04-07 19:49:37.000000000 -0400
+++ polly-3.6.0.src/lib/CMakeLists.txt	2015-04-07 19:54:59.000000000 -0400
@@ -20,6 +20,13 @@
        CodeGen/PTXGenerator.cpp)
 endif (GPU_CODEGEN)
 
+set(system_libs)
+if(LLVM_ENABLE_TERMINFO)
+  if(HAVE_TERMINFO)
+    set(system_libs ${system_libs} ${TERMINFO_LIBS})
+  endif()
+endif()
+  
 add_polly_library(Polly
   Analysis/Dependences.cpp
   Analysis/ScopDetection.cpp
@@ -74,6 +81,7 @@
 # that static library.
 add_polly_loadable_module(LLVMPolly
   Polly.cpp
+  LINK_LIBS ${system_libs}
 )
 
 if (TARGET intrinsics_gen)
