diff -uNr gcc-5.0-20141109/gcc/ipa-chkp.c gcc-5.0-20141109.fink/gcc/ipa-chkp.c
--- gcc-5.0-20141109/gcc/ipa-chkp.c	2014-11-07 08:56:40.000000000 -0500
+++ gcc-5.0-20141109.fink/gcc/ipa-chkp.c	2014-11-10 10:57:05.000000000 -0500
@@ -41,7 +41,6 @@
 #include "cgraph.h"
 #include "tree-chkp.h"
 #include "ipa-chkp.h"
-#include <string>
 
 /*  Pointer Bounds Checker has two IPA passes to support code instrumentation.
 
diff -uNr gcc-5.0-20141109/gcc/system.h gcc-5.0-20141109.fink/gcc/system.h
--- gcc-5.0-20141109/gcc/system.h	2014-11-09 10:40:28.000000000 -0500
+++ gcc-5.0-20141109.fink/gcc/system.h	2014-11-10 10:57:05.000000000 -0500
@@ -194,6 +194,13 @@
 #undef fread_unlocked
 #undef fwrite_unlocked
 
+/* Include <string> before "safe-ctype.h" to avoid GCC poisoning
+   the ctype macros through safe-ctype.h */
+
+#ifdef __cplusplus
+# include <string>
+#endif
+
 /* There are an extraordinary number of issues with <ctype.h>.
    The last straw is that it varies with the locale.  Use libiberty's
    replacement instead.  */
diff -uNr gcc-5.0-20141109/gcc/wide-int.h gcc-5.0-20141109.fink/gcc/wide-int.h
--- gcc-5.0-20141109/gcc/wide-int.h	2014-10-29 19:57:27.000000000 -0400
+++ gcc-5.0-20141109.fink/gcc/wide-int.h	2014-11-10 10:57:05.000000000 -0500
@@ -2129,7 +2129,7 @@
 inline WI_BINARY_RESULT (T1, T2)
 wi::smin (const T1 &x, const T2 &y)
 {
-  return min (x, y, SIGNED);
+  return wi::min (x, y, SIGNED);
 }
 
 /* Return the minimum of X and Y, treating both as unsigned values.  */
@@ -2137,7 +2137,7 @@
 inline WI_BINARY_RESULT (T1, T2)
 wi::umin (const T1 &x, const T2 &y)
 {
-  return min (x, y, UNSIGNED);
+  return wi::min (x, y, UNSIGNED);
 }
 
 /* Return the maxinum of X and Y, treating them both as having
@@ -2160,7 +2160,7 @@
 inline WI_BINARY_RESULT (T1, T2)
 wi::smax (const T1 &x, const T2 &y)
 {
-  return max (x, y, SIGNED);
+  return wi::max (x, y, SIGNED);
 }
 
 /* Return the maximum of X and Y, treating both as unsigned values.  */
@@ -2168,7 +2168,7 @@
 inline WI_BINARY_RESULT (T1, T2)
 wi::umax (const T1 &x, const T2 &y)
 {
-  return max (x, y, UNSIGNED);
+  return wi::max (x, y, UNSIGNED);
 }
 
 /* Return X & Y.  */
diff -uNr gcc-5.0-20141109/libcc1/plugin.cc gcc-5.0-20141109.fink/libcc1/plugin.cc
--- gcc-5.0-20141109/libcc1/plugin.cc	2014-11-07 08:56:40.000000000 -0500
+++ gcc-5.0-20141109.fink/libcc1/plugin.cc	2014-11-10 10:57:05.000000000 -0500
@@ -55,8 +55,6 @@
 #include "connection.hh"
 #include "rpc.hh"
 
-#include <string>
-
 #ifdef __GNUC__
 #pragma GCC visibility push(default)
 #endif
