--- gcc-5.0-20141106/libcc1/plugin.cc	2014-11-07 08:56:40.000000000 -0500
+++ gcc-5.0-20141106/libcc1/plugin.cc.new	2014-11-07 08:46:02.000000000 -0500
@@ -32,6 +32,7 @@
 #undef PACKAGE_VERSION
 
 #include "gcc-plugin.h"
+#include <string>
 #include "system.h"
 #include "coretypes.h"
 #include "stringpool.h"
@@ -55,8 +56,6 @@
 #include "connection.hh"
 #include "rpc.hh"
 
-#include <string>
-
 #ifdef __GNUC__
 #pragma GCC visibility push(default)
 #endif
