diff --git a/gcc/cgraphunit.c b/gcc/cgraphunit.c
index 6f61f5c..89c96dc 100644
--- a/gcc/cgraphunit.c
+++ b/gcc/cgraphunit.c
@@ -2342,6 +2342,14 @@ cgraph_node::create_wrapper (cgraph_node *target)

     cgraph_edge *e = create_edge (target, NULL, 0, CGRAPH_FREQ_BASE);

+    tree arguments = DECL_ARGUMENTS (decl);
+
+    while (arguments)
+      {
+       TREE_ADDRESSABLE (arguments) = false;
+       arguments = TREE_CHAIN (arguments);
+      }
+
     expand_thunk (false, true);
     e->call_stmt_cannot_inline_p = true;


