--- gcc-5.0-20141106/gcc/cfgexpand.c.orig	2014-11-07 06:47:52.000000000 -0500
+++ gcc-5.0-20141106/gcc/cfgexpand.c	2014-11-07 06:48:45.000000000 -0500
@@ -1722,6 +1722,9 @@
 
   init_vars_expansion ();
 
+  if (targetm.use_pseudo_pic_reg ())
+    pic_offset_table_rtx = gen_reg_rtx (Pmode);
+
   hash_map<tree, tree> ssa_name_decls;
   for (i = 0; i < SA.map->num_partitions; i++)
     {
--- gcc-5.0-20141106/gcc/function.c.orig	2014-11-07 06:48:53.000000000 -0500
+++ gcc-5.0-20141106/gcc/function.c	2014-11-07 06:50:03.000000000 -0500
@@ -3674,11 +3674,6 @@
 
   fnargs.release ();
 
-  /* Initialize pic_offset_table_rtx with a pseudo register
-     if required.  */
-  if (targetm.use_pseudo_pic_reg ())
-    pic_offset_table_rtx = gen_reg_rtx (Pmode);
-
   /* Output all parameter conversion instructions (possibly including calls)
      now that all parameters have been copied out of hard registers.  */
   emit_insn (all.first_conversion_insn);
