diff -ruN boost_1_46_1-orig/tools/build/v2/tools/darwin.jam boost_1_46_1/tools/build/v2/tools/darwin.jam
--- boost_1_46_1-orig/tools/build/v2/tools/darwin.jam	2011-07-02 14:47:42.000000000 +0900
+++ boost_1_46_1/tools/build/v2/tools/darwin.jam	2011-07-03 10:07:15.000000000 +0900
@@ -20,6 +20,8 @@
 import property-set ;
 import regex ;
 import errors ;
+import os ;
+import option ;
 
 ## Use a framework.
 feature framework : : free ;
@@ -247,9 +249,9 @@
         # Set the flags the version needs to compile with, first
         # generic options.
         flags darwin.compile OPTIONS $(condition)/<macosx-version>$(version-feature)
-            : -isysroot $(sdk) ;
+            : -d $(sdk) ;
         flags darwin.link OPTIONS $(condition)/<macosx-version>$(version-feature)
-            : -isysroot $(sdk) ;
+            : -d $(sdk) ;
         
         # Then device variation options.
         switch $(version[1])
@@ -536,11 +538,16 @@
 {
     setup-address-model $(targets) : $(sources) : $(properties) ;
     prepare-framework-path $(<) ;
+    local prefix = [ option.get prefix ] ; 
+    LIB_INSTALL_DIR = [ option.get libdir : $(prefix)/lib ] ;
+    VERSIONINFO_COMP = [ option.get compatibility_version ] ; 
+    VERSIONINFO_CUR = [ option.get current_version ] ; 
+     UNDEFINED = [ option.get undefined ] ;
 }
 
 actions link.dll bind LIBRARIES
 {
-    "$(CONFIG_COMMAND)" -dynamiclib -Wl,-single_module -install_name "$(<:B)$(<:S)" -L"$(LINKPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -l$(FINDLIBS-SA) -l$(FINDLIBS-ST) $(FRAMEWORK_PATH) -framework$(_)$(FRAMEWORK:D=:S=) $(OPTIONS) $(USER_OPTIONS)
+    "$(CONFIG_COMMAND)" -dynamiclib -Wl,-single_module $(UNDEFINED) -install_name $(<:D=$(LIB_INSTALL_DIR):S=.dylib) -compatibility_version 1.46.0 -current_version 1.46.1 -L"$(LINKPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -l$(FINDLIBS-SA) -l$(FINDLIBS-ST) $(FRAMEWORK_PATH) -framework$(_)$(FRAMEWORK:D=:S=) $(OPTIONS) $(USER_OPTIONS)
 }
 
 # We use libtool instead of ar to support universal binary linking
