diff -Naur octave-3.0.5/src/DLD-FUNCTIONS/__glpk__.cc octave-3.0.5.patched//src/DLD-FUNCTIONS/__glpk__.cc
--- octave-3.0.5/src/DLD-FUNCTIONS/__glpk__.cc	2009-01-16 01:46:23.000000000 -0500
+++ octave-3.0.5.patched//src/DLD-FUNCTIONS/__glpk__.cc	2010-09-18 19:13:15.000000000 -0400
@@ -59,8 +59,8 @@
 
 #else
 
-void _glp_lib_print_hook (int (*func)(void *info, char *buf), void *info);
-void _glp_lib_fault_hook (int (*func)(void *info, char *buf), void *info);
+//void _glp_lib_print_hook (int (*func)(void *info, char *buf), void *info);
+//void _glp_lib_fault_hook (int (*func)(void *info, char *buf), void *info);
 
 #endif
 }
@@ -137,15 +137,15 @@
 
 static jmp_buf mark;  //-- Address for long jump to jump to
 
-int
-glpk_fault_hook (void * /* info */, char *msg)
+ int
+glpk_error_hk (void * /* info */, char *msg)
 {
   error ("CRITICAL ERROR in GLPK: %s", msg);
   longjmp (mark, -1);
-}
+} 
 
 int
-glpk_print_hook (void * /* info */, char *msg)
+glpk_term_hk (void * /* info */, char *msg)
 {
   message (0, "%s", msg);
   return 1;
@@ -165,16 +165,16 @@
   clock_t t_start = clock();
 
 #ifdef GLPK_PRE_4_14
-  lib_set_fault_hook (0, glpk_fault_hook);
+  lib_set_fault_hook (0, glpk_error_hk);
 #else
-  _glp_lib_fault_hook (glpk_fault_hook, 0);
+  glp_term_hook (glpk_term_hk, 0);
 #endif
 
   if (lpxIntParam[0] > 1)
 #ifdef GLPK_PRE_4_14
     lib_set_print_hook (0, glpk_print_hook);
 #else
-    _glp_lib_print_hook (glpk_print_hook, 0);
+//    _glp_lib_print_hook (glpk_print_hook, 0);
 #endif
 
   LPX *lp = lpx_create_prob ();
@@ -318,7 +318,7 @@
       insist (method != method);
 #else
       static char tmp[] = "method != method";
-      glpk_fault_hook (0, tmp);
+      //glpk_fault_hook (0, tmp);
 #endif
     }
 
